### PR TITLE
feat: adding fcc news fetch project

### PR DESCRIPTION
**Pull request by jdwilkin4**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

## Live link
https://hc5t7v.csb.app/

## Topics covered

- strings
- arrays
- getElementById
- const and let
- fetch
- promises
- catch block
- slice
- console.error
- arrow functions
- template literals
- if statements
- textContent
- click events
- addEventListener
- object destructuring
- innerHTML
- forEach
- string concatenation
-

### CODE REVIEW

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const authorContainer = document.getElementById("author-container");
+const loadMoreBtn = document.getElementById("load-more-btn");
+let startingIndex = 0;
+let endingIndex = 8;
+let authorDataArr;
+
+// replace this API with one hosted by FCC
+fetch("https://fcc-author-api.herokuapp.com/authors")
```
Comment (naomi-lgbt) [inline Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: Would you like to open a PR to add the API code to https://github.com/freeCodeCamp/demo-projects?

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const authorContainer = document.getElementById("author-container");
+const loadMoreBtn = document.getElementById("load-more-btn");
+let startingIndex = 0;
+let endingIndex = 8;
+let authorDataArr;
```
Comment (naomi-lgbt) [inline Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:5]:
```suggestion
let authorDataArr = [];
```

I'm personally not a fan of declaring without initialising (TypeScript would get angry that we're using `slice` on a value that might be `undefined`).

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    displayUsers(authorDataArr.slice(startingIndex, endingIndex));
+  })
+  .catch((err) => {
+    authorContainer.innerHTML = `<p class="error-msg">There was an error loading the data</p>`;
+    console.error(`There was an error: ${err}`);
+  });
+
+const fetchMoreAuthors = () => {
+  startingIndex += 8;
+  endingIndex = endingIndex + 8;
```
Comment (naomi-lgbt) [inline Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:21]: Is there a specific reason not to use addition assignment like we did above?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Coming along for my usual nitpicking

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 1 above)
Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: I opened a PR here
https://github.com/freeCodeCamp/demo-projects/pull/368

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 3 above)
Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:21]: Yeah, that is a good point. 
I can change that real quick 👍

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 1 above)
Comment (scissorsneedfoodtoo) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: Thanks @jdwilkin4, just looked at your PR in the Demo Projects repo and everything looks good at first glance.

Before we review and merge that, though, would it be simpler to host this JSON file in the CDN repo instead? My understanding is that we won't update it much if at all once it's hosted somewhere.

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>freeCodeCamp Author page</title>
```
Comment (scissorsneedfoodtoo) [inline Learn-JSON-Fetch-async-await-building-fcc-author-page/index.html:7]: Very minor suggestion:

```suggestion
    <title>freeCodeCamp News Author Page</title>
```

But the current `title` works, too.

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 1 above)
Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: We shouldn't need to update it at all.
As long as the campers can still fetch the data from there, then hosting it on the CDN is fine with me.
Should I go ahead and close my demo projects PR?
Which one is the CDN repo?

Comment (scissorsneedfoodtoo) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: Hi @jdwilkin4, sorry, just seeing this now.

Awesome. thanks for confirming that the data will be static.

Yes, I think the PR in the demo projects repo can be closed.

Here's the link to the CDN repo: https://github.com/freeCodeCamp/cdn

I think we can keep things simple and just create a new directory in `/build/curriculum`, name it something like `news-author-page`, and put the JSON file there.

Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: @scissorsneedfoodtoo 

I created the PR here
https://github.com/freeCodeCamp/cdn/pull/181

Comment (scissorsneedfoodtoo) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]: Thanks for adding that JSON file to the CDN repo, @jdwilkin4, and for your review and merge, @naomi-lgbt.

I just swapped the Heroku URL out for the [one served by our CDN](https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json) and everything LGTM.

**Comment by scissorsneedfoodtoo** [comment]
@jdwilkin4, took another look locally and everything LGTM.

I think the last quick thing to do is to pull in all the recent changes Naomi made to the directory structure and project names, and place this project in the `javascript-algorithms-and-data-structures` directory.

I don't know if we landed on a final title for this project, but we could probably go with something like `learn-fetch-and-promises-by-building-an-fcc-authors-page` or `learn-basic-fetch-and-promises-by-building-an-fcc-authors-page` for this project's directory.

Once that's finished, this should be good to merge and to start breaking down into steps.

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 1 above)
Comment (naomi-lgbt) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js:8]:
```suggestion
fetch("https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json")
```

Looks like this still needs to be updated?

**Comment by jdwilkin4** [comment]
@scissorsneedfoodtoo 
I like `learn-fetch-and-promises-by-building-an-fcc-authors-page`. I will update that 👍

File: Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js (hunk 1 above)
Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js]: @naomi-lgbt 

Ahh yes. 
Let me update that real quick

Comment (jdwilkin4) [reply Learn-JSON-Fetch-async-await-building-fcc-author-page/index.js]: updated 👍

**Review by naomi-lgbt — APPROVED** [review]
This should be good to go now.

**Pull request merged**